kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/bytes-simplification.k

```diff
@@ -36,6 +36,8 @@ module BYTES-SIMPLIFICATION [symbolic]
 
     // #buf
 
+    rule [buf-empty]: #buf(N:Int, _) => b"" requires N ==Int 0 [simplification]
```

Comment:
This indeed seems to fix the issue. @PetarMax, do you think it's possible to rewrite the definition of `#buf` so that this simplification is not necessary: https://github.com/runtimeverification/evm-semantics/blob/2e060f15e32dfa02d924d32eb058ab7c27a31a24/kevm-pyk/src/kevm_pyk/kproj/evm-semantics/buf.md?plain=1#L52

Reply:
I am not sure, all of these function evaluation rules were designed for concreteness. When symbolic values come in, then we use simplifications.

Reply:
Alright, makes sense. One last thing: do we understand the root cause of this regression?